### DISCUSSION TITLE
Add validation in SQLModel.sqlmodel_update

### DISCUSSION THREAD

**Discussion by MatthewMckee4** [fastapi/sqlmodel#1241] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Self

from pydantic import model_validator
from sqlmodel import Field, SQLModel


class Person(SQLModel):
    age: int
    name: str
    has_job: bool


class Child(Person):
    name: str = Field(primary_key=True)

    @model_validator(mode="after")
    def validate_age(self) -> Self:
        if self.age < 18 and self.has_job:
            raise ValueError("Age must be greater than 18")
        return self


child = Child(age=19, name="John", has_job=True)

update_data = {"age": 17}

child.sqlmodel_update(update_data) # Raise no error here
```

### Description

This is very short example, but here when i update the child with invalid data, it does not do any sort of validation on the updated model, this results in an invalidated instance.

It would be nice if it re-validated the class after updating it

### Operating System

Linux

### Operating System Details

Ubuntu 24.04

### SQLModel Version

.0.22

### Python Version

3.12.3

**Comment by monchin** [comment]
It seems simillar to the issue that sqlmodel actually does not use pytantic to validate when you set `table=True`. Maybe `sqlmodel_update` does not enable pydantic validation either.

**Comment by adonh** [comment]
have same request.
following.

**Comment by YuriiMotov** [comment]
Code example can be simplified:

```py
from sqlmodel import SQLModel, Field


class Child(SQLModel):
    age: int = Field(..., lt=18)
    height: float


child = Child(age=17, height=1.5)

# Try updating with invalid data
child.sqlmodel_update({"age": "not an int", "height": "not a float"})  # DID NOT RAISE ...

print(child)
# age='not an int' height='not a float'

```
